core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -173,7 +183,23 @@ public synchronized void connect() throws IOException, ConnectionFailedException
           new TBinaryProtocol(mTransportProvider.getClientTransport(mAddress));
       mProtocol = new TMultiplexedProtocol(binaryProtocol, getServiceName());
       try {
-        mProtocol.getTransport().open();
+        // Set a timeout for transport.open() in case it hangs when the authentication type does not
+        // match between Alluxio client and servers.
+        FutureTask<Void> openTask = new FutureTask<>(
+            new TransportOpenCallable(mProtocol.getTransport()));
+        executor.submit(openTask);
+        try {
+          openTask.get((long) socketTimeoutMs, TimeUnit.MILLISECONDS);
```

Comment:
This doesn't really timeout the for connection open request. Why do you still need this given we have a timeout in TransportProvierUtils#createThriftSocket which times out for creating a socket.